src/store/modules/designs.ts

```diff
@@ -40,7 +41,73 @@ export default {
       axios
         .get(`${settings.apis.designStorage}/designs`)
         .then((response: AxiosResponse<DesignItem[]>) => {
-          commit("setDesigns", response.data);
+          const promises = response.data.map(design => {
+            // if design comes initially from job then full reaction information
+            // is already stored in the design storage
+            if (design.method === "Pathway") {
+              return {
+                ...design,
+                design: {
+                  ...design.design,
+                  added_reactions: design.design.reaction_knockins.map(
+                    reaction => JSON.parse(reaction)
+                  )
+                }
+              };
+            } else {
+              // otherwise request full information from BIGG
+              return axios
+                .all(
+                  design.design.reaction_knockins.map(reactionId =>
+                    axios.get(
+                      `${
+                        settings.apis.bigg
+                      }/models/universal/reactions/${reactionId}`
+                    )
+                  )
+                )
+                .then(response => {
+                  let addedReactions: any = [];
+                  response.forEach(r => {
+                    const reaction = {
+                      id: r.data.bigg_id,
+                      name: r.data.name,
+                      reactionString: r.data.reaction_string,
+                      lowerBound: -1000,
+                      upperBound: 1000,
+                      gene_reaction_rule: "",
+                      metabolites: {
+                        ...r.data.metabolites.map(m => ({
+                          [`${m.bigg_id}_${
+                            m.compartment_bigg_id
+                          }`]: m.stoichiometrys
+                        }))
+                      },
+                      metabolites_to_add: r.data.metabolites.map(m => ({
```

Comment:
We shouldn't need this now - it'd be enough to add metabolites with the interface from the interactive map store, then this will be automatically mapped to escher/cobra format when adding the reaction.